zebra-consensus/src/block/check.rs

```diff
@@ -184,14 +185,42 @@ pub fn subsidy_is_valid(
         // Note: Canopy activation is at the first halving on mainnet, but not on testnet
         // ZIP-1014 only applies to mainnet, ZIP-214 contains the specific rules for testnet
         // funding stream amount values
-        let funding_streams = zebra_chain::parameters::subsidy::funding_stream_values(
+        let mut funding_streams = zebra_chain::parameters::subsidy::funding_stream_values(
             height,
             network,
             expected_block_subsidy,
         )
         // we always expect a funding stream hashmap response even if empty
         .map_err(|err| BlockError::Other(err.to_string()))?;
 
+        let has_expected_output = |address, expected_amount| {
+            subsidy::funding_streams::filter_outputs_by_address(coinbase, address)
+                .iter()
+                .map(zebra_chain::transparent::Output::value)
+                .any(|value| value == expected_amount)
+        };
+
+        // The deferred pool contribution is checked in `miner_fees_are_valid()`
+        // See [ZIP-1015](https://zips.z.cash/zip-1015) for more details.
+        let mut deferred_pool_balance_change = funding_streams
+            .remove(&FundingStreamReceiver::Deferred)
+            .unwrap_or_default()
+            .constrain::<NegativeAllowed>()
+            .expect("should be valid Amount");
+
+        // TODO: Add references to the one-time lockbox disbursement & community coinholder funding model ZIPs
+        //       (https://zips.z.cash/draft-ecc-lockbox-disbursement, https://zips.z.cash/draft-ecc-community-and-coinholder)
+        let expected_one_time_lockbox_disbursements = network.lockbox_disbursement(height);
+        for (address, expected_amount) in &expected_one_time_lockbox_disbursements {
+            if !has_expected_output(address, *expected_amount) {
+                Err(SubsidyError::OneTimeLockboxDisbursementNotFound)?;
+            }
```

Comment:
This isn't sufficient; we need an exact match between disbursements and outputs, not simply set inclusion of the expected (address, output amount) pairs.